waiter/integration/waiter/metrics_output_test.clj

```diff
@@ -140,3 +140,63 @@
         (is (>= (get-in aggregate-metrics ["counters" "request-counts" "total"]) num-requests)))
 
       (delete-service waiter-url service-id))))
+
+
+(defmacro get-p-value
+  [metric p]
+  `(let [p# ~p
+         p-value# (get-in ~metric ["value" p#])]
+     (is (number? p-value#) (str "missing p" p# " value"))
+     p-value#))
+
+(deftest ^:parallel ^:integration-slow test-launch-metrics-output
+  (testing-using-waiter-url
+    (let [waiter-settings (waiter-settings waiter-url)
+          metrics-sync-interval-ms (get-in waiter-settings [:metrics-config :metrics-sync-interval-ms])
+          router->endpoint (routers waiter-url)
+          router-urls (vals router->endpoint)
+          service-name (rand-name)
+          sleep-seconds 20
+          min-startup-seconds 10 ; 20s +/- 10s for 2 polls with 5s granularity
+          max-startup-seconds 60 ; the service shouldn't take more than a minute to become healthy
+          instance-count 2
+          req-headers {:x-waiter-cmd (str "sleep " sleep-seconds " && " (kitchen-cmd "-p $PORT0"))
+                       :x-waiter-cmd-type "shell"
+                       :x-waiter-min-instances instance-count
+                       :x-waiter-name service-name}
+          {:keys [headers request-headers service-id] :as first-response}
+          (make-request-with-debug-info req-headers #(make-kitchen-request waiter-url % :method :get))]
+      (with-service-cleanup
+        service-id
+        ; ensure the first request succeded before continuing with testing
+        (assert-response-status first-response 200)
+
+        ; ensure each router has had a chance to publish its local metrics
+        (Thread/sleep 10000)
```

Comment:
Can this be written using `wait-for` pattern, e.g. wait for the launch metrics entry for the service to show up on each router from the `/state/launch-metrics` endpoint?

Reply:
Done.